packages/bpk-component-page-indicator/src/BpkPageIndicator.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { cssModules } from '../../bpk-react-utils';
+
+import NavButton, { DIRECTIONS } from './NavButton';
+import STYLES from './BpkPageIndicator.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const DISPLAYED_TOTAL = 5;
+const START_SCROLL_INDEX = Math.floor(DISPLAYED_TOTAL / 2);
+
+export const INDICATOR_STYLE_TYPE = {
+  default: 'default',
+  overImage: 'overImage',
+};
+
+export type Props = {
+  className: ?string,
+  showNav: ?boolean,
+  currentIndex: number,
+  totalIndicators: number,
+  onClick: ?() => void,
+  indicatorLabel: string,
+  prevNavLabel: string,
+  nextNavLabel: string,
+  style: ?$Keys<typeof INDICATOR_STYLE_TYPE>,
+};
+
+const BpkPageIndicator = ({
+  className,
+  currentIndex,
+  indicatorLabel,
+  nextNavLabel,
+  onClick,
+  prevNavLabel,
+  showNav,
+  style,
+  totalIndicators,
+}: Props) => (
+  // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
+  <div className={className}>
+    <div
+      className={getClassName(
+        'bpk-page-indicator',
+        showNav && 'bpk-page-indicator__showNav',
+      )}
+    >
+      {showNav && (
+        <NavButton
+          currentIndex={currentIndex}
+          onClick={onClick}
+          disabled={currentIndex === 0}
+          prev
+          ariaLabel={prevNavLabel}
+        />
+      )}
+      <div className={getClassName('bpk-page-indicator__container')}>
+        <div
+          className={getClassName('bpk-page-indicator__indicators-container')}
+          style={
+            currentIndex > START_SCROLL_INDEX
+              ? {
+                  '--scroll-index':
+                    totalIndicators > DISPLAYED_TOTAL
+                      ? Math.min(
+                          currentIndex - START_SCROLL_INDEX,
+                          totalIndicators - DISPLAYED_TOTAL,
+                        )
+                      : 0,
+                }
+              : undefined
+          }
+        >
+          {[...Array(totalIndicators)].map((val, index) => (
+            <button
+              type="button"
```

Comment:
Just wondering, is it a requirement from design for the indicators to be clickable? They seem to be quite small clicking targets, so I guess it's for keyboard navigation only?

Reply:
Because it is considered that they will also be used on the desktop, and there are many places in projects where there is a need to click, so it is set here to be clickable. 
This also provides better accessibility and allows users to select the page number you want to go to within a certain range via the keyboard.